src/main/java/io/strimzi/kafka/bridge/tracing/OpenTelemetryHandle.java

```diff
@@ -0,0 +1,213 @@
+import org.apache.kafka.clients.producer.ProducerConfig;
+
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.COMPONENT;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.KAFKA_SERVICE;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.OPENTELEMETRY_SERVICE_NAME_ENV_KEY;
+
+/**
+ * OpenTelemetry implementation of Tracing.
+ *
+ * Note: we use Vert.x OpenTelemetry extension to setup custom ContextStorageProvider:
+ * @see io.vertx.tracing.opentelemetry.VertxContextStorageProvider
+ */
+class OpenTelemetryHandle implements TracingHandle {
+
+    private Tracer tracer;
+    private ExecutorService service;
+
+    static void setCommonAttributes(SpanBuilder builder, RoutingContext routingContext) {
+        builder.setAttribute(SemanticAttributes.PEER_SERVICE, KAFKA_SERVICE);
+        builder.setAttribute(SemanticAttributes.HTTP_METHOD, routingContext.request().method().name());
+        builder.setAttribute(SemanticAttributes.HTTP_URL, routingContext.request().uri());
+    }
+
+    @Override
+    public String envServiceName() {
+        return OPENTELEMETRY_SERVICE_NAME_ENV_KEY;
+    }
+
+    @Override
+    public String serviceName(BridgeConfig config) {
+        return System.getenv(envServiceName());
+    }
+
+    @Override
+    public void initialize() {
+        System.setProperty("otel.metrics.exporter", "none"); // disable metrics
+        AutoConfiguredOpenTelemetrySdk.initialize();
+    }
+
+    @Override
+    public synchronized ExecutorService adapt(ExecutorService provided) {
+        if (service == null) {
+            service = Context.taskWrapping(provided);
+        }
+        return service;
+    }
+
+    private Tracer get() {
+        if (tracer == null) {
+            tracer = GlobalOpenTelemetry.getTracer(COMPONENT);
+        }
+        return tracer;
+    }
+
+    private SpanBuilder getSpanBuilder(RoutingContext routingContext, String operationName) {
+        Tracer tracer = get();
+        SpanBuilder spanBuilder;
+        Context parentContext = propagator().extract(Context.current(), routingContext, ROUTING_CONTEXT_GETTER);
+        if (parentContext == null) {
+            spanBuilder = tracer.spanBuilder(operationName);
+        } else {
+            spanBuilder = tracer.spanBuilder(operationName).setParent(parentContext);
+        }
+        return spanBuilder;
+    }
+
+    @Override
+    public <K, V> SpanBuilderHandle<K, V> builder(RoutingContext routingContext, String operationName) {
+        SpanBuilder spanBuilder = getSpanBuilder(routingContext, operationName);
+        return new OTelSpanBuilderHandle<>(spanBuilder);
+    }
+
+    @Override
+    public <K, V> void handleRecordSpan(SpanHandle<K, V> parentSpanHandle, KafkaConsumerRecord<K, V> record) {
+        String operationName = String.format("%s %s", record.topic(), MessageOperation.RECEIVE);
```

Comment:
Can we change this to use string concat. `String.format` is surprisingly expensive, it's not providing much value here and it appears to be on a relatively hot path (the POLL request)